### PR TITLE
Linux + v8

### LINKED ISSUE #426: [CI] Linux/Clang Validation Tests timeout
Only on Linux with Clang, the CI tests time out.
Re-running the failed task just work fine. 
Find out why and fix it.

### PR DESCRIPTION
**Pull request by CedricGuillemet**
After some tests and experiments on Linux, I've realized 2 things:
- v8-dev package exists 
- validation tests with v8 work fine

fixes #426 
Something is wrong with JSC Ubuntu. It might be our impl, JSC package, buggy JSC version, ...
Add V8 by default and Validation tests for most linux jobs, Add 1 JSC+GCC that doesn't run validation tests. 
Fixed few warning pragmas for V8. Some methods changed between 2 V8 versions, preprocessor check added to take care of it.

### CODE REVIEW

File: Dependencies/napi/CMakeLists.txt [hunk 1]
```diff
@@ -7,8 +7,10 @@ if (WIN32)
     set(NAPI_JAVASCRIPT_ENGINE "Chakra" CACHE STRING "JavaScript engine for N-API.")
 elseif (ANDROID)
     # engine is selected in native build cmakelists.txt
-elseif (APPLE OR UNIX)
+elseif (APPLE)
     set(NAPI_JAVASCRIPT_ENGINE "JavaScriptCore" CACHE STRING "JavaScript engine for N-API.")
+elseif (UNIX)
+    # engine is selected by cmake command line
```
Comment (bghgary) [inline Dependencies/napi/CMakeLists.txt:13]: This just sets the default. If the cmake command line sets one, it should override this.

File: Dependencies/napi/napi-direct/CMakeLists.txt [hunk 2]
```diff
@@ -72,10 +72,18 @@ if(NOT TARGET javascript_engine)
...
-        target_link_to_dependencies(javascript_engine INTERFACE ${JSCORE_LIBRARY})
+    elseif(UNIX AND NOT ANDROID AND NOT APPLE)
+        if(NAPI_JAVASCRIPT_ENGINE STREQUAL "JavaScriptCore")
+            target_include_directories(napi PUBLIC "/usr/include/webkitgtk-4.0")
+            find_library(JSCORE_LIBRARY JavaScriptCore)
+            target_link_to_dependencies(javascript_engine INTERFACE ${JSCORE_LIBRARY})
+        elseif(NAPI_JAVASCRIPT_ENGINE STREQUAL "V8")
+            target_include_directories(napi PUBLIC "/usr/include/v8")
+            find_library(V8_LIBRARY v8)
+            target_link_to_dependencies(javascript_engine INTERFACE ${V8_LIBRARY})
```
Comment (bghgary) [inline Dependencies/napi/napi-direct/CMakeLists.txt:83]: Do we want this define in V8 for linux like the one in Windows?
`V8_COMPRESS_POINTERS`

I'm not sure it's better to split by OS all of a sudden here. Would it better to split against JS engine and then split by OS if needed?

File: Dependencies/napi/napi-direct/source/js_native_api_v8.cc [hunk 3]
```diff
@@ -3036,10 +3036,16 @@ napi_status napi_detach_arraybuffer(napi_env env, napi_value arraybuffer) {
...
   RETURN_STATUS_IF_FALSE(
       env, it->IsDetachable(), napi_detachable_arraybuffer_expected);
 
   it->Detach();
+#else
+  RETURN_STATUS_IF_FALSE(
+      env, it->IsNeuterable(), napi_detachable_arraybuffer_expected);
 
+  it->Neuter();
+#endif
```
Comment (bghgary) [inline Dependencies/napi/napi-direct/source/js_native_api_v8.cc:3049]: Did you find out why the linux version is using an old version of V8? How does node get away with this?

File: README.md [hunk 4]
```diff
@@ -342,13 +342,33 @@ The minimal requirement target is an OpenGL 3.3 compatible GPU. Clang 9+ or GCC
...
+```
+cmake -GNinja -DJSCORE_LIBRARY=/usr/lib/x86_64-linux-gnu/libjavascriptcoregtk-4.0.so  -DNAPI_JAVASCRIPT_ENGINE=JavaScriptCore ..
+```
+
+For V8:
 
 ```
-cmake -GNinja -DJSCORE_LIBRARY=/usr/lib/x86_64-linux-gnu/libjavascriptcoregtk-4.0.so ..
+cmake -GNinja -DNAPI_JAVASCRIPT_ENGINE=V8 ..
 ```
```
Comment (bghgary) [inline README.md:372]: Might be clearer if you bucket the build steps together per JS engine. This is somewhat hard to read IMO. You are also missing a `For JavaScriptCore:` in the second step but will be moot if we group by JS engine.

**Review by bghgary — APPROVED** [review]

File: Dependencies/napi/napi-direct/source/js_native_api_v8.cc (hunk 3 above)
Comment (CedricGuillemet) [reply Dependencies/napi/napi-direct/source/js_native_api_v8.cc:3049]: Let me check  with latest Ubuntu. My packages are not latest on my Linux box

Comment (CedricGuillemet) [reply Dependencies/napi/napi-direct/source/js_native_api_v8.cc:3049]: I've added a note here : https://github.com/BabylonJS/BabylonNative/issues/791

File: README.md (hunk 4 above)
Comment (CedricGuillemet) [reply README.md]: Indeed. Fixed by splitting in 2 subsections.

File: Dependencies/napi/napi-direct/CMakeLists.txt (hunk 2 above)
Comment (CedricGuillemet) [reply Dependencies/napi/napi-direct/CMakeLists.txt]: `V8_COMPRESS_POINTERS` might be great for Linux, issue is determining when building for 64bits targets with CMake. I take a note .

**Pull request merged**